latex/ledger-spec.tex

```diff
@@ -387,84 +407,97 @@ \subsection{Deposits and Refunds}
+number associated with the author's key in the resource allocation argument.
+
+The function $\fun{keyRefunds}$, in turn, uses $\fun{certRefund}$ to calculate
+the total value to be refunded to all individual key deregistration
+certificate authors in a transaction.
+Given protocol constants, resource allocations, and a transaction,
+this calculation sums up all the refunds for the individual key
+deregistration certificates
+carried by the given transaction by passing the relevant parameters to the
+$\fun{certRefund}$ function.
+
+It is important to note here that instead of the \textit{current} slot number,
+the time to live of $\var{tx}$ is passed to the $\fun{certRefunds}$ function
+within the summation in $\fun{keyRefunds}$. The reason for this is that the
+refunds for any key deregistration certificates are, in fact, included in
+the $\var{tx}$ itself - meaning that the coin value of the refund must be
+explicitly specified in the outputs of the transaction. So,
+the value of the included refund must be calculated before this transaction
+is ever processed, and be the same \textit{no matter when} $\var{tx}$
+\textit{is actually processed} in order to for the system to continue to
+satisfy the general accounting property.
+
+It is impossible to predict the exact slot
+number in which $\var{tx}$ will be processed, but it will be some time before
+slot number $\fun{ttl}~\var{tx}$. So, this is the slot number value used in both
+the calculation to generate the refund coin value in the outputs of $\var{tx}$
+and in the general accounting property equation.
+
+Note that since the refund is based
+on the original deposit paid (as indicated in the protocol constants that
+were valid in the time slot the deposit was made), the accounting or the slot
+number do not also need to be updated in any way when updating pool constants with
+a certificate.
+
+Note also that
+$\fun{keyRefunds}$ calculates the total individual refunds for a transaction
+based on \textit{current} protocol constants. This means that any deposits
+made prior to a change will be different from their corresponding
+(decayed) refunds in the case of key deregistration after a change in
+protocol constants.
+The protocol constants are not
+expected to change often, and using the current ones for the calculation
+is a deliberate simplification choice, which does not introduce any inconsistencies
+into the system rules or properties. In particular, the general accounting
+property is not violated.
+
+Finally, note that that a refund for a resource-releasing certificate in a
 transaction can only be issued if the refund is bigger than the minimum
 transaction fee. In order to receive a refund in this case, the transaction
 body must also contain inputs onto which this refund can be added.
 
+\begin{todo}
+Is it necessary to have the releasing c check here? Since the refund calc is
+only summed over either keyreg or poolreg certificates?
```

Comment:
This case is to ensure that the function is total. It's true that in this spec we only ever call it for certs `c` where `releasing c` is true, but for the function takes a `DCert` so we need this case for totality.